apps/mini-runtime/src/main/java/com/akto/hybrid_parsers/HttpCallParser.java

```diff
@@ -166,7 +170,24 @@ public void syncFunction(List<HttpResponseParams> responseParams, boolean syncIm
             syncTrafficMetricsWithDB();
             this.last_synced = Context.now();
             this.sync_count = 0;
-            MergeLogicLocal.mergingJob(apiCatalogSync.dbState);
+            /*
+             * submit a job only if it is not running.
+             */
+            if (!pgMerging) {
+                int accountId = Context.accountId.get();
+                pgMerging = true;
+                service.submit(() -> {
+                    Context.accountId.set(accountId);
```

Comment:
Write entire thread inside a try-catch and set pgMerging to false inside a finally block